From 0f74f431e0970a2e5784dbd955cfa4760e3b1ef7 Mon Sep 17 00:00:00 2001
From: James Holmes <jdh@rtds.com>
Date: Sun, 29 Aug 2021 21:54:01 -0500
Subject: [PATCH] Fixed major security bug.

A user could effectively specify an undefined url and the program would return a null filter rather than throwing an exception and returning an error to the user. This was a really bad bug. I found it when I was adding code to validate the path parameters. That will be coming in the next commit.
---
 src/main/java/com/rtds/svc/CaptureTypeService.java | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/main/java/com/rtds/svc/CaptureTypeService.java b/src/main/java/com/rtds/svc/CaptureTypeService.java
index 52bdea3..be6c346 100644
--- a/src/main/java/com/rtds/svc/CaptureTypeService.java
+++ b/src/main/java/com/rtds/svc/CaptureTypeService.java
@@ -52,12 +52,16 @@ public String findFilter( String url_suffix )
         
         CaptureType type = em.find( CaptureType.class, url_suffix );
         
-        if( type != null )
+        if( type == null )
         {
-            return type.getCaptureFilter();
+            throw new IllegalArgumentException( "The url_suffix must exist in the database." );
         }
         
-        return null;
+        // It is okay for the capture filter itself to be null, but the CaptureType
+        // must be in the database, otherwise the user could effectively forge
+        // a capture filter for "all" just by requesting a non-existent filter.
+        
+        return type.getCaptureFilter();
     }
     
     public CaptureType find( String url_suffix )
